src/utils/cache.h

```diff
@@ -186,121 +207,99 @@ class LruCache {
     SpinMutex::Lock lock(mutex_);
     return size_;
   }
-  int GetCapacity() const { 
-	return capacity_.load(std::memory_order_relaxed);
-  }
-  static constexpr size_t GetItemStructSize() { return sizeof(Item); }
+  int GetCapacity() const { return capacity_.load(std::memory_order_relaxed); }
+  static constexpr size_t GetItemStructSize() { return sizeof(Entry); }
 
  private:
-  struct Item {
-    Item(K key, std::unique_ptr<V> value)
+  struct Entry {
+    Entry() {}
+    Entry(uint64_t key, std::unique_ptr<V> value)
         : key(key), value(std::move(value)) {}
-    K key;
+    uint64_t key;
     std::unique_ptr<V> value;
     int pins = 0;
-    Item* next_in_hash = nullptr;
-    Item* prev_in_queue = nullptr;
-    Item* next_in_queue = nullptr;
+    // 1 for populated.
+    int metadata = 0;
```

Comment:
Why not `bool` or `enum`?
And probably something like `is_used` is better than `metadata`.

Reply:
Yeah this evolved - it was a tri state and I was just being lazy.  Now a bool seems fine - will change.

Reply:
Done.